vlasovsolver/gpu_acc_map.cpp

```diff
@@ -970,19 +975,37 @@ __global__ void __launch_bounds__(WID3,Hashinator::defaults::MAX_BLOCKSIZE/(WID3
    const vmesh::VelocityMesh* __restrict__ vmesh = vmeshes[cellOffset];
    vmesh::VelocityBlockContainer *blockContainer = blockContainers[cellOffset];
    Realf *gpu_blockData = blockContainer->getData();
-   const Realf minValue = (Realf)dev_minValues[cellOffset];
+
+   // Load minvalues to shared memory
+   __shared__ Realf minValue;
+   __shared__ uint setColumnOffset;
+   __shared__ uint numColumns;
 
    // shared memory buffer for reducing looping count per block
-   __shared__ int loopN[WID3];
+   __shared__ int loopN[WID3/GPUTHREADS];
 
-   if (setIndex >= columnData->dev_sizeColSets()) {
-      return;
+   {
```

Comment:
Oh, hm, would this be useful in other kernels too, reducing the scope of certain elements where possible, freeing up registers etc?